### DISCUSSION TITLE
[Feature Request]: Allow configuring post type registration with_front to false

### DISCUSSION THREAD

**Discussion by mdavoli** [code4recovery/12-step-meeting-list#964] [Ideas]
### Contact Details

no response

### Requested Feature/Enhancment

Currently the plugin registers all post types with the with_front configuration parameter set to true. It would be nice if there was a way to change this to false via a filter or some other configuration. Otherwise if the blogs url is not configured to default then the url will not be at root of the website which is preferred.

See documentation here: https://developer.wordpress.org/reference/functions/register_post_type/#:~:text=to%20%24post_type%20key.-,with_front,-bool

### TSML Version

Latest (default)

### Wordpress Version

Latest (default)

**Comment by mdavoli** [comment]
You can see here: https://staging.orlandorecovery.com/recovery-blog/meetings/ how it's tacking /meetings/ to the blog url and not to the root of the site.

**Reply by tim-rohrer** [reply]
Hi Mark—

Thanks for opening the topics. I converted them because we ask non-C4R members to use Discussions first, and then we on the team will create any issues.

While I believe I understand what your proposal would do, I don’t understand why this is useful, at least for 99.9% of our users. Can you expand on why this would be useful/essential?

Thanks!

**Reply by mdavoli** [reply]
The benefit is merely one of customization. Right now there is no possible way to change the root url if your blog url is configured in permalinks to be other than the root directory which forces the url to be something like `/<blog_url>/meetings/` when it would be a nice feature to have a filter that would allow you to override the post registration or change the `with_front` configuration to false. 

The only other option would be for us to edit the plugin files which would make updating the plugin more challenging in the future.

**Reply by 97eighty** [reply]
@mdavoli good argument. Do you have this working now? Are you able to submit PR?

**Reply by mdavoli** [reply]
@97eighty  pull request made: https://github.com/code4recovery/12-step-meeting-list/pull/966

**Comment by 97eighty** [comment]
A filter would be a good solution. Like Tim said, most would not use but adding a filter around the rewrite array would allow you to override - https://developer.wordpress.org/reference/functions/register_taxonomy
